lib/cdo/user_helpers.rb

```diff
@@ -20,7 +20,7 @@ def self.generate_username(queryable, name)
     return prefix if queryable.where(username: prefix).limit(1).empty?
 
     # Throw darts to find an appropriate suffix, using it if we hit bullseye.
-    (0..1).each do |exponent|
+    (0..2).each do |exponent|
       min_index = 10**exponent
       max_index = 10**(exponent + 1) - 1
       3.times do |_i|
```

Comment:
Gotcha. I stand corrected. 50% more darts, with the new round hopefully having more chance of success than the previous rounds.

The question is, is that enough to drop the # of fallback calls to near zero. My input to the conversation is I would still err on the side of more darts. The indexed query is going to be way, way faster than the fallback (which I think turns into a full table scan of 14M rows?). I wouldn't blink at trying up to 20 indexed queries to avoid the fallback.